### DISCUSSION TITLE
Different Outputs and Pre-SMO Residual in Old and New MOOSE Application

### DISCUSSION THREAD

**Discussion by shrituntunroy** [idaholab/moose#28462] [Q&A General] [answered]
Hi Moose Team,

I am new to MOOSE and currently in the process of learning how to use it.

I am working with a MOOSE-based application, Rhocp (https://github.com/apatra6/rhocp). When I run MOOSE input files using this Rhocp application, the command line output looks like this:

```bash
Time Step 121, time = 7.03415, dt = 0.1
Pre-SMO residual: 3.60514e-05
  Applying predictor with scale factor = 1
[32m Solve Converged![39m
  Finished Solving                                                                       [[33m  6.56 s[39m] [[33m  143 MB[39m]

Time Step 122, time = 7.13415, dt = 0.1
Pre-SMO residual: 3.6051e-05
  Applying predictor with scale factor = 1
[32m Solve Converged![39m
  Finished Solving                    
```

Since this Rhocp application uses an older MOOSE standard, I created a new Rhocp application using `~/projects/moose/scripts/stork.sh Rhocp` and copied the necessary files (materials, auxkernels, userobjects, etc.) from the original version.

However, when I run the same input file using this new Rhocp-opt application, the command line output is different:

```bash
Time Step 11, time = 0.00321504, dt = 0.000619174
  Applying predictor with scale factor = 1
[32m Solve Converged![39m

Time Step 12, time = 0.00395805, dt = 0.000743008
  Applying predictor with scale factor = 1
[32m Solve Converged![39m

Time Step 13, time = 0.00484966, dt = 0.00089161
  Applying predictor with scale factor = 1
[32m Solve Converged![39m

Time Step 14, time = 0.00591959, dt = 0.00106993
  Applying predictor with scale factor = 1
[32m Solve Converged![39m
```

I have the following questions:

1. Why am I seeing different command line outputs when using the older Rhocp version from [this repository](https://github.com/apatra6/rhocp) compared to the new Rhocp version created using `stork.sh`, even though I copied the same materials, auxkernels, and other files?

2. What is the "Pre-SMO residual" displayed in the older version's output? How can I enable or use this feature in the new Rhocp application I created?

I have tried changing the following line in RhocpApp.C in the base folder

```C++
  InputParameters params = MooseApp::validParams();
  params.set<bool>("use_legacy_material_output") = false;
  params.set<bool>("use_legacy_initial_residual_evaluation_behavior") = true; // from false
```

still I am getting the same command line output with this new RhoCp based app.

Further I have noticed that for some problems the convergence with old rhocp-opt is better than new rhocp-opt. I think may be reason is this Pre-SMO residual

Thank you for your help!

**Comment by GiudGiud** [comment, marked as answer]
Hello

First the default output of residuals might have changed. You will want to turn on more output of residuals in the Outputs block. See the CommonOutputAction for the list of parameters 

The pre-SMO residual was an early residual evaluation that was not always necessary. Selecting the legacy behavior should re-establish that residual evaluation. 

You can use Debug/show_execution_order=ALWAYS to see this initial residual evaluation happen. 

Guillaume

**Comment by shrituntunroy** [comment]
Thankyou [Giudicelli](https://github.com/GiudGiud) for your help
